### PR TITLE
popup install scala plugin window if choose create scala project

### PR DESCRIPTION
**Pull request by ziyel**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/SparkScalaSettingsStep.java [hunk 1]
```diff
@@ -56,6 +56,9 @@ public void updateDataModel() {
 
     @Override
     public boolean validate() throws ConfigurationException {
+        if (this.builder.getScalaPluginStatus() != ScalaPluginStatus.INSTALLED) {
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/SparkScalaSettingsStep.java]: can combine this with the following return:
```java
return (this.builder.getScalaPluginStatus() == ScalaPluginStatus.INSTALLED) && super.validate() && (javaStep == null || javaStep.validate()); 
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/HDInsightModuleBuilder.java [hunk 2]
```diff
@@ -136,23 +163,43 @@ public void moduleCreated(@NotNull Module module) {
...
-    private void addTelemetry(HDInsightTemplatesType templatesType, SparkVersion sparkVersion){
+    private void showScalaPluginInstallMsg() {
+        JLabel label = new JLabel(SCALA_PLUGIN_INSTALL_MSG);
+        label.setOpaque(false);
+        label.addMouseListener(new MouseAdapter() {
+            @Override
+            public void mouseReleased(MouseEvent e) {
+                Set<String> pluginIds = new HashSet<>();
+                pluginIds.add(SCALA_PLUGIN_ID);
+                PluginsAdvertiser.installAndEnablePlugins(pluginIds, () -> scalaPluginStatus = ScalaPluginStatus.NEED_RESTART);
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/HDInsightModuleBuilder.java]: the lambda looks weird

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/HDInsightModuleBuilder.java [hunk 3]
```diff
@@ -47,19 +50,30 @@
...
 
 public class HDInsightModuleBuilder extends JavaModuleBuilder implements ModuleBuilderListener {
     private HDInsightTemplateItem selectedTemplate;
     private LibrariesContainer librariesContainer;
     private LibraryCompositionSettings scalaLibraryCompositionSettings;
     private SparkVersion sparkVersion;
+    private ScalaPluginStatus scalaPluginStatus;
+
+    private static final String SCALA_PLUGIN_ID = "org.intellij.scala";
+    private static final String SCALA_PLUGIN_INSTALL_MSG = "<HTML>No Scala Intellij plugin found. " +
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/HDInsightModuleBuilder.java]: The html looks complex, can we use the code to generate it? such as the following example:

```java
Document makeBasicHtmlDoc(String title) {
        Document htmlDoc = new MyHTMLDocumentImpl();
        DocumentType docType = new DocumentTypeImpl(null, "html",
                "-//W3C//DTD XHTML 1.0 Strict//EN",
                "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd");
        htmlDoc.appendChild(docType);
        Element htmlElement = htmlDoc.createElementNS("http://www.w3.org/1999/xhtml", "html");
        htmlDoc.appendChild(htmlElement);
        Element headElement = htmlDoc.createElement("head");
        htmlElement.appendChild(headElement);
        Element titleElement = htmlDoc.createElement("title");
        if(title != null)
            titleElement.setTextContent(title);
        headElement.appendChild(titleElement);
        Element bodyElement = htmlDoc.createElement("body");
        htmlElement.appendChild(bodyElement);

        return htmlDoc;
    }
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/HDInsightModuleBuilder.java (hunk 2 above)
Comment (ziyel) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/HDInsightModuleBuilder.java]: Yes, it's confusing. Have updated it. Thanks.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/SparkScalaSettingsStep.java (hunk 1 above)
Comment (ziyel) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/SparkScalaSettingsStep.java]: The combined return statement is a bit complex. Make the `if (this.builder.getScalaPluginStatus() != ScalaPluginStatus.INSTALLED)` in a single line, we can emphasize that if the scala plugin not ready, we can't create the project.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/HDInsightModuleBuilder.java (hunk 3 above)
Comment (ziyel) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/HDInsightModuleBuilder.java]: Thanks for the suggestion. The `html` in `String` is more readable. The `SCALA_PLUGIN_INSTALL_MSG ` is not dynamic. We use the html to emphasize the `click` word in the string. Use `String` is enough. Let's keep it.

**Pull request merged**